src/target/stm32_common.c

```diff
@@ -51,6 +51,10 @@ bool stm32_uid(target_s *const target, const target_addr_t uid_base)
 	uid.wafer_ycoord = read_le2(uid_bytes, 2);
 	uid.wafer_number = uid_bytes[4];
 	memcpy(uid.lot_number, &uid_bytes[5], 7);
+	/* Avoid decoding as non-printable characters */
+	for (size_t j = 5; j < 12; j++)
```

Comment:
Missing `U` suffixes, and braces around the for block.

Reply:
Made everything unsigned, added braces around if-return.